data_structures/linked_list/singly_link_list_deletion.c

```diff
@@ -18,38 +19,75 @@ struct node *createnode()  // function to create node
     return (t);
 }
 ////////////////////////////////////////////////////////
-void insert()  // function to insert at first location
+int insert(int pos)  // function to insert at first location
```

Comment:
Not sure there's much difference between `void` and `int` as you're returning `0` with `int.`. IMO, we should use `void`.

Reply:
But as I am returning the function in certain cases(when insertion is not possible in input locations) in line 47,I need to use "int" to do that else segmentation fault occurs in case of "void"